### PR TITLE
Docs: Update block & lua references

### PR DESCRIPTION
**Pull request by TekMath**

### CODE REVIEW

File: lua/docs/content/reference/index.yml [hunk 1]
```diff
@@ -60,6 +60,7 @@ blocks:
     - list:
         - "<a href=\"/reference/array\">array</a>"
         - "<a href=\"/reference/boolean\">boolean</a>"
+        - "<a href=\"/reference/condition\">condition</a>"
```
Comment (gdevillele) [inline lua/docs/content/reference/index.yml:63]: `condition` is not a Lua data type 🤔 ,
but this PR is a nice addition.

would you accept to add it in a new list below ?
like so :

```yaml
    - text: "Lua concepts:"
     - list:
         - "<a href=\"/reference/condition\">condition</a>"
```

**Review by gdevillele — COMMENTED** [review]
Thank you! 
It looks good to me, but I requested a small change before we can accept this PR.

File: lua/docs/content/reference/index.yml (hunk 1 above)
Comment (TekMath) [reply lua/docs/content/reference/index.yml:63]: Yes no problem! it's my bad, I had misunderstood the type of category that it was.
We could add the loop reference in this new part

Comment (gdevillele) [reply lua/docs/content/reference/index.yml:63]: Thanks.
Indeed. 🙂

**Review by gdevillele — APPROVED** [review]
LGTM 👍

**Pull request merged**